File: app/src/main/java/com/pgssoft/mvvm/services/interfaces/MapperService.java
package com.pgssoft.mvvm.services.interfaces;

import com.pgssoft.mvvm.model.api.ApiRate;
import com.pgssoft.mvvm.model.api.ApiTable;
import com.pgssoft.mvvm.model.database.Rate;
import com.pgssoft.mvvm.model.database.Table;
import com.pgssoft.mvvm.models.dto.RateDto;

/**
 * Created by bstokrocki on 31.01.2017.
 */
public interface MapperService {
    Rate map(ApiRate apiRate);

    Table map(ApiTable apiTable);

    RateDto map(Rate rate);
}

File: kotlin/app/src/main/java/com/pgssoft/kotlinplayground/service/iface/MapperService.kt
package com.pgssoft.kotlinplayground.service.iface

import com.pgssoft.kotlinplayground.model.RateDto
import com.pgssoft.kotlinplayground.model.api.ApiRate
import com.pgssoft.kotlinplayground.model.api.ApiTable
import com.pgssoft.kotlinplayground.model.db.Rate
import com.pgssoft.kotlinplayground.model.db.Table

/**
 * Created by bstokrocki on 31.01.2017.
 */
interface MapperService {
    fun map(apiRate: ApiRate): Rate

    fun map(apiTable: ApiTable): Table

    fun map(rate: Rate): RateDto
}
